daprdocs/content/en/getting-started/quickstarts/pubsub-quickstart.md

```diff
@@ -0,0 +1,605 @@
+
+When you run `dapr init`, Dapr creates a default redis `pubsub.yaml` on your local machine. Verify by opening your components directory:
+
+- On Windows, under `%UserProfile%\.dapr\components\pubsub.yaml`
+- On Linux/MacOS, under `~/.dapr/components/pubsub.yaml`
+
+For this quickstart, we've included a default Redis `pubsub.yaml` file within the cloned repository that contains the following:
+
+{{< tabs "Self-Hosted (CLI)" "Kubernetes" >}}
+
+{{% codetab %}}
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: order_pub_sub
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+scopes:
+  - orderprocessing
+  - checkout
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+Run `kubectl apply -f pubsub.yaml` to apply the following:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: order_pub_sub
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: redis-master.default.svc.cluster.local:6379
+  - name: redisPassword
+    value: ""
+scopes:
+  - orderprocessing
+  - checkout
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+### Step 3: Subscribe to topics
+
+Navigate to the directory containing your subscriber application.
+
+```bash
+cd pub_sub/csharp/CheckoutService/Controllers
+```
+
+With the following command, run the CheckoutServiceController.cs service alongside a Dapr sidecar.
+
+```bash
+dapr run --app-id checkout --app-port 6002 --dapr-http-port 3602 --dapr-grpc-port 60002 --app-ssl --components-path ../../components dotnet run
+```
+
+In the CheckoutServiceController.cs subscriber, we're subscribing to the Redis instance called `order_pub_sub` (as defined in the `pubsub.yaml` component) and topic `orders`.
+```cs
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System;
+using Microsoft.AspNetCore.Mvc;
+using Dapr;
+using Dapr.Client;
+
+namespace CheckoutService.controller
+{
+    [ApiController]
+    public class CheckoutServiceController : Controller
+    {
+        [Topic("order_pub_sub", "orders")]
+        [HttpPost("checkout")]
+        public void getCheckout([FromBody] int orderId)
+        {
+            Console.WriteLine("Subscriber received : " + orderId);
+        }
+    }
+}
+```
+This enables your app code to talk to the Redis component instance through the Dapr sidecar. With the `pubsub.yaml` component, you can easily swap out underlying components without application code changes.
+
+### Step 4: Publish a topic
+
+Navigate to the directory holding the "Order Processing Service" Program.cs publisher.
+
+```bash
+cd pub_sub/csharp/OrderProcessingService
+```
+
+With the following command, run Program.cs service alongside a Dapr sidecar.
+
+```bash
+dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 --app-ssl --components-path ../../components dotnet run
```

Comment:
Same here - if app protocol is grpc, i'm not sure dapr-http-port is required. Instead, I would imagine dapr-grpc-port is needed